src/main/java/com/google/sps/utils/UserUtils.java

```diff
@@ -153,6 +158,29 @@ public static void addPoints(Entity userEntity, int numPoints, DatastoreService
     datastore.put(userEntity);
   }
 
+  /**
+    gets the users of a particular game to populate the leaderboard
+  */
+  public static ArrayList<User> userList(String gameId, DatastoreService datastore){
+    //create and prepare a query
+    Filter queryFilter = new FilterPredicate("gameId", FilterOperator.EQUAL, gameId);
+    Query query = new Query("user").setFilter(queryFilter).addSort("score", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    
+    // stores each user in a User object 
+    ArrayList<User> users = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        String userID = (String) entity.getProperty("userID");
+        String userName = (String) entity.getProperty("username");
+        int score = ((Number) entity.getProperty("score")).intValue();
+    
+        User user = new User(userID, userName, score);
+        users.add(user);
```

Comment:
Most of the value of having a `User` class is to encapsulate the user `Entity` object. If you have a `User` but there's code outside of `User` to convert `User` to and from `Entity`, then `User` is not making your life any easier.

I recommend an approach where an `Entity` is the sole private variable inside `User`. Then you can create a `User` around an entity and manipulate the entity by calling the setters and getters on `User`.

Reply:
Alright cool I changed most of it to fit the syntax of your code for it; with the exception of a few extra parameters to be gotten by the javascript for the leaderboard